src/mesh/distributed_mesh.C

```diff
@@ -58,7 +58,30 @@ DistributedMesh::DistributedMesh (const Parallel::Communicator & comm_in,
   _partitioner = libmesh_make_unique<ParmetisPartitioner>();
 }
 
+DistributedMesh & DistributedMesh::operator= (DistributedMesh && other_mesh)
+{
+  // Move assign as an UnstructuredMesh.
+  this->UnstructuredMesh::operator=(std::move(other_mesh));
+
+  // Nodes and elements belong to ReplicatedMesh and have to be
+  // moved before we can move arbitrary GhostingFunctor, Partitioner,
+  // etc. subclasses.
+  this->move_nodes_and_elements(std::move(other_mesh));
+
+  _extra_ghost_elems = std::move(other_mesh._extra_ghost_elems);
+
+  // Handle remaining MeshBase moves.
+  this->post_dofobject_moves(std::move(other_mesh));
+
+  return *this;
+}
 
+MeshBase & DistributedMesh::assign(MeshBase && other_mesh)
+{
+  *this = std::move(*(cast_ptr<DistributedMesh*>(&other_mesh)));
```

Comment:
`cast_ref<DistributedMesh&>(other_mesh)`

seems clearer (assuming it compiles?) rather than going from pointer back to reference.